plugins/woocommerce-blocks/assets/js/blocks/collection-filters/inner-blocks/rating-filter/edit.tsx

```diff
@@ -0,0 +1,337 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Icon, chevronDown } from '@wordpress/icons';
+import classnames from 'classnames';
+import { useBlockProps, InspectorControls } from '@wordpress/block-editor';
+import type { BlockEditProps } from '@wordpress/blocks';
+import Rating, {
+	RatingValues,
+} from '@woocommerce/base-components/product-rating';
+import {
+	useQueryStateByKey,
+	useQueryStateByContext,
+	useCollectionData,
+} from '@woocommerce/base-context/hooks';
+import { getSettingWithCoercion } from '@woocommerce/settings';
+import { isBoolean, isObject, objectHasProp } from '@woocommerce/types';
+import { useState, useMemo, useEffect } from '@wordpress/element';
+import { CheckboxList } from '@woocommerce/blocks-components';
+import FormTokenField from '@woocommerce/base-components/form-token-field';
+import type { ReactElement } from 'react';
+import {
+	Disabled,
+	Notice,
+	PanelBody,
+	ToggleControl,
+	withSpokenMessages,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { previewOptions } from './preview';
+import './style.scss';
+import { Attributes } from './types';
+import { formatSlug, getActiveFilters, generateUniqueId } from './utils';
+import { useSetWraperVisibility } from '../../../filter-wrapper/context';
+import './editor.scss';
+import { Inspector } from './components/inspector';
+
+const NoRatings = (
+	<Notice status="warning" isDismissible={ false }>
+		<p>
+			{ __(
+				"Your store doesn't have any products with ratings yet. This filter option will display when a product receives a review.",
+				'woocommerce'
+			) }
+		</p>
+	</Notice>
+);
```

Comment:
This notice shows up after adding the block to the page even there are some products with rating. The notice disappear after saving and reloading the editor.

![image](https://github.com/woocommerce/woocommerce/assets/5423135/a1689461-a33e-4229-a8d7-85b4ba70fc85)

Reply:
🤔 thanks I'll investigate

Reply:
@dinhtungdu I can't reproduce this. Could you give exact steps?

Reply:
I can't reproduce it now either. I'm marking this as resolved.